app/services/service_offering/add_to_portfolio_item.rb

```diff
@@ -27,25 +25,22 @@ def process
 
     private
 
-    def determine_valid_fields
-      # These are the fields that topology gave us, so we want to pull them into our model.
-      # Along the way we map them to their string counterparts, then reject the columsn we don't need
-      @service_offering.instance_variables
-                       .map { |var| var.to_s.delete("@") }
-                       .reject { |field| IGNORE_FIELDS.include?(field) }
+    def creation_fields
+      service_offering_columns = @service_offering.instance_variables.map { |x| x.to_s[1..-1] }
```

Comment:
@gmcculloug - There is no `column_names` or `columns` method available as checked [here](https://gist.github.com/syncrou/c3ddfd9cdb7f70c6810110842a3f3aed) so I went with the most efficient method I could find 😉 I left in all the other tests added before so the class is fully tested.  But this code is now largely back to where it was when we started the conversation.

Reply:
Looks good.

FYI - You do not need the `to_s` call here as it will work with a symbol.  
```ruby
irb(main):006:0> :@name[1..-1]
=> "name"
```